src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -207,24 +207,38 @@ public long countSearchGermplasm(final GermplasmSearchRequest germplasmSearchReq
 	}
 
 	@Override
-	public PedigreeDTO getPedigree(final Integer germplasmDbId, final String notation, final Boolean includeSiblings) {
+	public PedigreeDTO getPedigree(final String germplasmDbId, final String notation, final Boolean includeSiblings) {
```

Comment:
1. Call this.germplasmValidator.validateGermplasmGUID and throw the exception right away. You can create a `validateGUID` method:
```
this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
		this.germplasmValidator.validateGermplasmGUID(errors, germplasmDbId);
		if (this.errors.hasErrors()) {
			throw new ResourceNotFoundException(this.errors.getAllErrors().get(0));
		}
```
2. Remove `isPresent` checking since you've already validated the GUID exists
3. Remove throwing of ApiRuntimeExeption("invalid germplasmDbId) 
4. Remove catching MiddlewareQueryException